### DISCUSSION TITLE
hetzner: Is 'Name' in `zone.Name` case-sensitive?

### DISCUSSION THREAD

**Discussion by rameri** [go-acme/lego#2258] [Q&A] [answered]
Hi!

I'm now facing error `hetzner: could not get zone for domain [my.own.domain.at.hetzner] not found` now, after getting certificates worked for quite some time.
Now I'm searching what could be the cause...

When I manually (curl) make a valid request to the API `/api/v1/zones` , I'm getting the expected result:
```
{
  "id": "VALIDREALIDVALIDREALID",
  "name": "at.hetzner",
  "ttl": 60,
  "registrar": "",
  "legacy_dns_host": "",
[...]
}
```

So, I wonder if why lego cannot find the zone? Is is because in the meantime 'Name' in `zone.Name` is now case-sensitive?

https://github.com/go-acme/lego/blob/2d880c2e6c2ddf48b3ac2a69a35a05ac7fe6eb11/providers/dns/hetzner/internal/client.go#L150

br
Reza

**Comment by ldez** [comment, marked as answer]
Hello,

Does your zone name contain uppercase characters?
If yes, why?

Does your zone name contain non-ASCII characters?

Another possibility is a problem with your records:
1. Do you have CNAME records?
2. Is the answer to SOA calls provided the right information?

Another possibility is a problem is a problem when filtering the zone: https://github.com/go-acme/lego/blob/2d880c2e6c2ddf48b3ac2a69a35a05ac7fe6eb11/providers/dns/hetzner/internal/client.go#L163

And, another possibility is a change inside Hetzner API, in this case, you should contact Hetzner.

**Reply by rameri** [reply]
Hello Idez,

Thank you very much for your answer!
Since I don't speak English well enough, my question wasn't quite precise enough. But I was on the wrong track anyway...
Your question about the SOA put me on the right track. Thanks again!

Until then I assumed that the provider only uses Hetzner's API, without any DNS queries, to determine the proper Hetzner “zone.id”.

However, in `func FindZoneByFqdn(fqdn string) (string, error) {}` I have now seen that the SOA is used via DNS to determine the name of the zone before the “Hetzner API” is queried.

br
Reza
